scripts/get_loc_info.py

```diff
@@ -0,0 +1,219 @@
+     # @param name the function name we used
+     # @return loc info if we find such function in debug info
+     # we treat loc as 1 if we hit some function does not contain body of function such as malloc func.
+    """
+    _, name_debug_regexp = get_regex_based_on_func_name(name)
+    next_function_debug_regexp = re.compile(r'distinct !DISubprogram')
+    for index, plain_d in enumerate(plain_data, start=1):
+        if name_debug_regexp.search(plain_d):
+            for indexp, plain_dp in enumerate(
+                    plain_data[index + 1:], start=index + 2):
+                if next_function_debug_regexp.search(plain_dp):
+                    loc = comput_loc_for_a_func(
+                        plain_data, index - 1, indexp - 1)
+                    return loc
+            return 1  # nd function treat as one line
+    raise Exception(f'the debug info of {name} must exists.')
+
+
+def get_next_func_call(data, name, index):
+    """
+     # This function compute a function call of a function based on given function name and index
+     # The function in our c file will have the following pattern:
+     #      call <blah blah> <func_name> <blah blah>
+     # Then it will find any function call within current function
+     # Finally we gather the function call info and return.
+     # @param data stored data from .ll file.
+     # @param name the function name we used
+     # @param index the index that we current pointed
+     # @return None if we hit the end of searching (when reach the pattern '}').
+     #          function name and debug function name and corresponded index on the data
+    """
+    func_call_regexp = re.compile(r'call.*@(?!llvm).*')
+    end_of_cur_func = re.compile(r'^}')
+    for index_next, line in enumerate(data[index:], start=index):
+        if end_of_cur_func.search(line):
+            return None, None, None  # end of searching...
+        if func_call_regexp.search(line):
+            func_name = line.split("@")[1].split("(")[0]
+            debug_name = func_name.split(
+                ".")[0] if "." in func_name else func_name
+            if func_name == name:
+                continue
+            return debug_name, func_name, index_next
+    return None, None, None
+
+
+def compute_loc_of_func(data, name, total_loc):
+    """
+     # This function compute line of code of each function started by main
+     # We treat declare function as one line of code, 
+     #      those function will be either simplified code or predefined code on seahorn
+     # The function recursively compute each called function on the harness
+     # Then it collect loc of current function and store it into total_loc
+     # @param data stored data from .ll file.
+     # @param name the function name we used
+     # @param total_loc the result of loc that we computed
+     # @return nothing, use call by ref.
+    """
+    name_regexp, _ = get_regex_based_on_func_name(name)
```

Comment:
judging from the debug message, we seem to count `loc` of a same function multiple times if its called multiple times, is that the behaviour we are going for? if not, we can use a dictionary of {name: loc} instead and tally up loc of each function exactly once.

Reply:
Maybe I need to discuss this with arie later. This is what I understand for now. I think the loc of each harness that we analyze is all the functions we used/called in the running time.

Reply:
I think you are right. I will change to avoid counting the same function multiple times.